react/features/file-sharing/reducer.ts

```diff
@@ -55,44 +47,16 @@ ReducerRegistry.register<IFileSharingState>('features/file-sharing',
         };
     }
 
-    case UPDATE_CONFERENCE_METADATA: {
-        const { metadata } = action;
-
-        if (!metadata) {
-            return state;
-        }
-
+    case FILE_LIST_RECEIVED: {
         const files = new Map();
 
-        for (const [ key, value ] of Object.entries(metadata)) {
-            if (key.startsWith(FILE_SHARING_PREFIX)) {
-                const fileId = key.substring(FILE_SHARING_PREFIX.length + 1);
-
-                files.set(fileId, value);
-            }
-        }
-
-        if (files.size === 0) {
-            return state;
-        }
-
-        const newFiles: Map<string, IFileMetadata> = new Map(state.files);
-
-        for (const [ key, value ] of files) {
-            // Deleted files will not have fileId.
-            if (!value.fileId) {
-                newFiles.delete(key);
-            } else {
-                newFiles.set(key, value);
-            }
-        }
-
-        if (isEqual(newFiles, state.files)) {
-            return state;
-        }
+        // convert the array to map we want in the state
+        action.files.forEach((f: IFileMetadata) => {
+            files.set(f.fileId, f);
```

Comment:
I think this breaks if we are uploading a file (but haven't updated the metadata yet) and we get a remote update.

Since at that moment we have the state only locally.

Reply:
I don't understand this. This is the list of files we receive when joining. There is no upload here.

Reply:
Nvm I thought we got updates in bulk, but we only get this the first time.